### PR TITLE
GEOMESA-899 AgeOff Iterator for 1.2

### PR DESCRIPTION
**Pull request by jahhulbert-ccri**
Signed-off-by: Andrew Hulbert <ahulbert@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/iterators/DtgAgeOffTest.scala [hunk 1]
```diff
@@ -0,0 +1,133 @@
+package org.locationtech.geomesa.accumulo.iterators
```
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/iterators/DtgAgeOffTest.scala]: yeah i need the header thing

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyAgeOffFilter.scala [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    spec = options.get(Options.Sft)
+    sft = IteratorCache.sft(spec)
+
+    val kryoOptions = if (sft.getSchemaVersion < 9) SerializationOptions.none else SerializationOptions.withoutId
+    val kryo = IteratorCache.serializer(spec, kryoOptions)
+    reusableSF = kryo.getReusableFeature
+  }
+
+  override def accept(k: Key, v: Value): Boolean = {
+    reusableSF.setBuffer(v.get)
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyAgeOffFilter.scala]: the feature id won't be set in later schema versions here FYI - might not matter

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/iterators/DtgAgeOffTest.scala [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+class DtgAgeOffTest extends Specification with TestWithDataStore {
+
+  sequential
+
+  override val spec = "some_id:String,dtg:Date,geom:Point:srid=4326"
+
+  "DTGAgeOff" should {
+
+    def configAgeOff(ads: AccumuloDataStore, days: Int): Unit = {
+      val is = new IteratorSetting(5, "ageoff", classOf[KryoDtgAgeOffIterator].getCanonicalName)
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/iterators/DtgAgeOffTest.scala]: do you know what the vis iterators, etc run at? I seem to recall anything lower than 20 being system level

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyAgeOffFilter.scala (hunk 2 above)
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyAgeOffFilter.scala]: bc its in the key i assume? i guess i could add an option to pull that out of the key if its there?

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyAgeOffFilter.scala]: yeah i think it's actually a bug throughout 1.2.x - we fixed it in 1.3. it rarely comes up b/c id filters get turned into ranges. and it wouldn't matter here unless you were trying to 'accept' based on id

File: geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/iterators/DtgAgeOffTest.scala (hunk 3 above)
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/iterators/DtgAgeOffTest.scala]: i don't think the vis iterators are in the config at all they just run no matter what. versioning is at 20.

Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/accumulo/iterators/DtgAgeOffTest.scala]: yeah i am currently running at at 5 in production seems ok. the prio is only for user iterators not system iterators it looks like.

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyAgeOffFilter.scala [hunk 4]
```diff
@@ -0,0 +1,86 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
```
Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyAgeOffFilter.scala]: we tweaked the copyright headers - you should be able to run `mvn license:format` to fix

**Pull request merged**